custom_components/ocpp/api.py

```diff
@@ -167,27 +168,30 @@ async def on_connect(self, websocket, path: str):
                 cp = self.charge_points[self.cpid]
                 await self.charge_points[self.cpid].reconnect(websocket)
         except Exception as e:
-            _LOGGER.info(f"Exception occurred:\n{e}")
+            _LOGGER.error(f"Exception occurred:\n{e}", exc_info=True)
+
         finally:
             self.charge_points[self.cpid].status = STATE_UNAVAILABLE
             _LOGGER.info(f"Charger {cp_id} disconnected from {self.host}:{self.port}.")
 
     def get_metric(self, cp_id: str, measurand: str):
         """Return last known value for given measurand."""
         if cp_id in self.charge_points:
-            return self.charge_points[cp_id].get_metric(measurand)
+            return self.charge_points[cp_id]._metrics[measurand].value
         return None
 
     def get_unit(self, cp_id: str, measurand: str):
         """Return unit of given measurand."""
         if cp_id in self.charge_points:
```

Comment:
As per comment above

Reply:
No. I've made metrics a `defaultdict`. As a result, any metric that is not (yet) defined will have default values (None)